src/components/KeyboardDismissingFlatList/index.js

```diff
@@ -1,58 +1,51 @@
-import React, {Component} from 'react';
+import React, {useRef, useEffect, useCallback} from 'react';
 import {FlatList, Keyboard} from 'react-native';
 import * as DeviceCapabilities from '../../libs/DeviceCapabilities';
 
-class KeyboardDismissingFlatList extends Component {
-    constructor(props) {
-        super(props);
+function KeyboardDismissingFlatList(props) {
+    const isScreenTouched = useRef(false);
 
-        this.touchStart = this.touchStart.bind(this);
-        this.touchEnd = this.touchEnd.bind(this);
-    }
+    const touchStart = () => {
+        isScreenTouched.current = true;
+    };
 
-    componentDidMount() {
+    const touchEnd = () => {
+        isScreenTouched.current = false;
+    };
```

Comment:
NAB, I thought maybe it is slightly more correct to have these methods inside the `useEffect()` as only the methods created on first render would ever get used by the event listeners there. Not sure if it matters too much though.